src/TSL/ModuloTheories/ConsistencyChecking.hs

```diff
@@ -2,42 +2,58 @@
+-- Description :  Adds partial semantics back to uninterpreted functions by checking which combinations
+--                of uninterpreted predicates are satisfiable.  To avoid an exponential blow‑up in SMT
+--                calls, we now cap the number of combinations examined.
 -- Maintainer  :  Wonhyuk Choi
 module TSL.ModuloTheories.ConsistencyChecking
-  ( generateConsistencyAssumptions,
-    consistencyDebug,
-    ConsistencyDebugInfo (..),
-  )
-where
-
-import Control.Monad.Trans.Except
-import TSL.Base.Ast
-  ( AstInfo (..),
-    SymbolInfo (..),
-    deduplicate,
-  )
-import TSL.Error (Error, errConsistency)
-import TSL.ModuloTheories.Debug (IntermediateResults (..))
-import TSL.ModuloTheories.Predicates
-  ( TheoryPredicate (..),
-    enumeratePreds,
-    pred2Smt,
-    pred2Tsl,
-    predInfo,
-    predTheory,
-  )
-import TSL.ModuloTheories.Solver (solveSat)
-import TSL.ModuloTheories.Theories
-  ( Theory,
-    TheorySymbol,
-    isUninterpreted,
-    smtSortDecl,
-    symbol2Smt,
-    symbolType,
-  )
+  ( generateConsistencyAssumptions
+  , consistencyDebug
+  , ConsistencyDebugInfo (..)
+  , config_MAX_CONSISTENCY_CHECKS   
+  ) where
+
+import           Control.Monad.Trans.Except
+import qualified Data.List                      as L
+import           Debug.Trace                    (trace)
+
+import           TSL.Base.Ast                   (AstInfo (..), SymbolInfo (..), deduplicate)
+import           TSL.Error                      (Error, errConsistency)
+import           TSL.ModuloTheories.Debug       (IntermediateResults (..))
+import           TSL.ModuloTheories.Predicates  (TheoryPredicate (..), enumeratePreds
+                                               , pred2Smt, pred2Tsl, predInfo, predTheory)
+import           TSL.ModuloTheories.Solver      (solveSat)
+import           TSL.ModuloTheories.Theories    (Theory, TheorySymbol, isUninterpreted
+                                               , smtSortDecl, symbol2Smt, symbolType)
+
+-- |Hard cap on how many predicate combinations we send to the SMT solver.
+--   This prevents run‑away exponential behaviour when many uninterpreted
+--   predicates appear in the spec.  Adjust as needed or expose it as a CLI flag.
+-- this number can be changed based on the problem, but even for simple synthesis problems the 
+-- number of predicates being checked was always become 2^n. This simplifies while getting the correct answer
+config_MAX_CONSISTENCY_CHECKS :: Int
+config_MAX_CONSISTENCY_CHECKS = 100
+
+
+generateConsistencyAssumptions
+  :: FilePath                  
+  -> [TheoryPredicate]        
+  -> [ExceptT Error IO String]
+generateConsistencyAssumptions path preds =
+  map (fmap fst . consistencyChecking path) limitedCombos
+  where
+    -- enumerate all combinations, then keep only the first N
```

Comment:
are these generated in order of importance? i mean, if we have predicates [0...N], are the first queries all about predicate 0, then all the queries are about 1, etc? I think this is fine for now, but maybe leave a TODO here